tango/__main__.py

```diff
@@ -656,9 +732,48 @@ def _run(
                     + click.style(f"{info.result_location}", bold=True, fg="green")
                 )
 
-        click_logger.info(
-            click.style("Finished run ", fg="green") + click.style(run.name, fg="green", bold=True)
-        )
+        if step_name is not None:
+            uncacheable_leaf_steps = step_graph.uncacheable_leaf_steps()
+            step = step_graph[step_name]
+            executor.execute_step(step, step_name in uncacheable_leaf_steps)
+            log_step_summary(step)
+            click_logger.info(
+                click.style("Finished run for step ", fg="green")
+                + click.style(step_name, fg="green", bold=True)
+                + click.style(" (", fg="green", bold=True)
+                + click.style(run.name, fg="green", bold=True)
+                + click.style(")", fg="green", bold=True)
+            )
+        else:
+            failed_steps = executor.execute_step_graph(step_graph, run_name=run.name)
```

Comment:
It seems weird to me that `execute_step_graph()` returns failed steps. That's what you need here, sure, but `execute_step_graph()` is an incredibly general function, so it's weird to tailor the return type to one use case. At least, it should return a tuple `(succeeded_steps, failed_steps, neither_steps)`.